lib/rules/prefer-destructuring.js

```diff
@@ -82,14 +84,39 @@ module.exports = {
         }
 
         /**
-         * Report that the given node should use destructuring
+         * Find the length to slice the nodeText and the accessor object since destructuring
+         * removes the accessor syntax from the right hand side of the AST
          *
-         * @param {ASTNode} reportNode the node to report
-         * @param {string} type the type of destructuring that should have been done
-         * @returns {void}
+         * @param {Token} lastToken the last token in the reportNode
+         * @param {SourceCode} sourceCode the source code object for simplicity
+         * @returns {Object} removalState the state from the right hand side to move/remove the text
          */
-        function report(reportNode, type) {
-            context.report({ node: reportNode, message: `Use ${type} destructuring` });
+        function findRightHandRemovalState(lastToken, sourceCode) {
+            let lengthToRemove, accessorToken;
+
+            if (lastToken) {
+
+                // Calls to the object with dot syntax
+                if (lastToken.type === "Identifier") {
+                    accessorToken = lastToken;
+                    const punctuator = sourceCode.getTokenBefore(lastToken);
+
+                    lengthToRemove = lastToken.range[1] - punctuator.range[0];
+
+                // Calls to the object with bracket syntax
+                } else if (lastToken.type === "Punctuator" && lastToken.value === "]") {
+                    accessorToken = sourceCode.getTokenBefore(lastToken);
+                    const closingBracket = sourceCode.getTokenBefore(accessorToken);
+
+                    lengthToRemove = lastToken.range[1] - closingBracket.range[0];
+                }
+            } else {
+
+                // Case for no trailing semicolon
+                lengthToRemove = lastToken.range[1] - lastToken.range[0];
```

Comment:
I'm confused by this logic. If `lastToken` is falsy, won't this cause an error to be thrown?